### ISSUE TITLE
Decimal of skeleton file

### ISSUE THREAD

**Issue by JiajunZhang-neu** [takashi310/VVD_Viewer#2]
Hi,
I loaded a skeleton file of a neuron with three decimals, vvdviewer didn't render it correctly but as a ball-like structure. I scale the skeleton 1000 times in R and then vvdviewer can render it correctly. So it seems that vvdviewer only takes the integer part of skeleton coordinates.

May I ask for some improvement on skeleton loading? Thank you in advance!

Best,
Jiajun

**Comment by jefferis** [comment]
Is it possible that the radius associated with the skeletons did not scale? Maybe you can check this.

**Comment by jefferis** [comment]
@lankiszhang to be more specific, there is a chance that this is an issue with scaling in R rather than with VVD. Specifically, to scale the radius of neurons, you must multiply by a 4 vector not just a scalar. Example below:

```r
> library(nat)
> n=Cell07PNs[[1]]
> n1000=n*1000
> head(n$x)
NULL
> head(n$d)
  PointNo Label        X        Y        Z    W Parent
1       1     2 186.8660 132.7093 88.20393 1.01     -1
2       2     2 187.3355 131.1558 90.59680 1.27      1
3       3     2 188.1165 130.2545 93.14326 1.14      2
4       4     2 188.4680 129.4757 94.97399 1.27      3
5       5     2 188.2875 128.7542 97.45621 1.27      4
6       6     2 188.2733 127.9567 99.16801 1.27      5
> head(n1000$d)
  PointNo Label        X        Y        Z    W Parent
1       1     2 186866.0 132709.3 88203.93 1.01     -1
2       2     2 187335.5 131155.8 90596.80 1.27      1
3       3     2 188116.5 130254.5 93143.26 1.14      2
4       4     2 188468.0 129475.7 94973.99 1.27      3
5       5     2 188287.5 128754.2 97456.21 1.27      4
6       6     2 188273.3 127956.7 99168.01 1.27      5
> n1000=n*rep(1000,4)
> head(n1000$d)
  PointNo Label        X        Y        Z    W Parent
1       1     2 186866.0 132709.3 88203.93 1010     -1
2       2     2 187335.5 131155.8 90596.80 1270      1
3       3     2 188116.5 130254.5 93143.26 1140      2
4       4     2 188468.0 129475.7 94973.99 1270      3
5       5     2 188287.5 128754.2 97456.21 1270      4
6       6     2 188273.3 127956.7 99168.01 1270      5
```

**Comment by JiajunZhang-neu** [comment]
Dear Greg,

Thank you for your kindly advice and the helpful example. I don’t have the R environment at home now, but I will check it once I get to my office tomorrow morning. Actually we have already looked into that $d in the neuronlist, but we just focused on XYZ and missed the W value. I will check that carefully tomorrow. Thank you again for your help!

Best,
Jiajun



发件人: Gregory Jefferis
发送时间: Freitag, 16. Oktober 2020 00:01
收件人: takashi310/VVD_Viewer
抄送: lankiszhang; Mention
主题: Re: [takashi310/VVD_Viewer] Decimal of skeleton file (#2)

@lankiszhang to be more specific, there is a chance that this is an issue with scaling in R rather than with VVD. Specifically, to scale the radius of neurons, you must multiply by a 4 vector not just a scalar. Example below:
> library(nat)
> n=Cell07PNs[[1]]
> n1000=n*1000
> head(n$x)
NULL
> head(n$d)
  PointNo Label        X        Y        Z    W Parent
1       1     2 186.8660 132.7093 88.20393 1.01     -1
2       2     2 187.3355 131.1558 90.59680 1.27      1
3       3     2 188.1165 130.2545 93.14326 1.14      2
4       4     2 188.4680 129.4757 94.97399 1.27      3
5       5     2 188.2875 128.7542 97.45621 1.27      4
6       6     2 188.2733 127.9567 99.16801 1.27      5
> head(n1000$d)
  PointNo Label        X        Y        Z    W Parent
1       1     2 186866.0 132709.3 88203.93 1.01     -1
2       2     2 187335.5 131155.8 90596.80 1.27      1
3       3     2 188116.5 130254.5 93143.26 1.14      2
4       4     2 188468.0 129475.7 94973.99 1.27      3
5       5     2 188287.5 128754.2 97456.21 1.27      4
6       6     2 188273.3 127956.7 99168.01 1.27      5
> n1000=n*rep(1000,4)
> head(n$d)
  PointNo Label        X        Y        Z    W Parent
1       1     2 186.8660 132.7093 88.20393 1.01     -1
2       2     2 187.3355 131.1558 90.59680 1.27      1
3       3     2 188.1165 130.2545 93.14326 1.14      2
4       4     2 188.4680 129.4757 94.97399 1.27      3
5       5     2 188.2875 128.7542 97.45621 1.27      4
6       6     2 188.2733 127.9567 99.16801 1.27      5
> head(n1000$d)
  PointNo Label        X        Y        Z    W Parent
1       1     2 186866.0 132709.3 88203.93 1010     -1
2       2     2 187335.5 131155.8 90596.80 1270      1
3       3     2 188116.5 130254.5 93143.26 1140      2
4       4     2 188468.0 129475.7 94973.99 1270      3
5       5     2 188287.5 128754.2 97456.21 1270      4
6       6     2 188273.3 127956.7 99168.01 1270      5
